client/src/services/check.ts

```diff
@@ -0,0 +1,26 @@
+import { checkType, IBadReview } from './../components/BadReview/BadReviewControllers';
+import globalAxios, { AxiosInstance } from 'axios';
+
+export class CheckService {
+  private axios: AxiosInstance;
+  private cache: Record<number, Record<checkType, IBadReview[]>>;
+
+  constructor() {
+    this.axios = globalAxios.create({ baseURL: `/api/` });
+    this.cache = {};
+  }
+
+  async getBadComments(taskId: number, type: checkType): Promise<IBadReview[]> {
+    if (this.cache?.[taskId]?.[type]) return this.cache[taskId][type];
```

Comment:
You can create separate function-wrapper that will create / return cache.
Just look at these functions: "getBadComments" and "getMaxScoreCheckers" they way too similar.

Reply:
+